src/release_archive.bzl

```diff
@@ -14,6 +14,37 @@
 
 """Rules to create a release archive"""
 
+load("@rules_java//java:java_binary.bzl", "java_binary")
+load("@with_cfg.bzl", "with_cfg")
+
+# The minimum --java_{tool_,}runtime_version supported by prebuilt Java tools.
+_MINIMUM_JAVA_RUNTIME_VERSION = 8
+
+# The minimum version of a java_toolchain's java_runtime supported by prebuilt Java tools.
+_MINIMUM_JAVA_COMPILATION_RUNTIME_VERSION = 11
```

Comment:
@cushon If you think that it's time for that, we could raise this to 17 or 21.

Reply:
That SGTM but I'll defer to @hvadehra, I'm not sure how common/supported it is to be overriding `java_toolchain.runtime_version` to older versions

Reply:
I think we might as well bump it up to 17. JavaBuilder packages error-prone 2.38 which I think won't work with < 17.